cmake/Platform/Initialization/LoadArduinoPlatformSettings.cmake

```diff
@@ -58,11 +58,20 @@ if (NOT ${SETTINGS_LIST} AND EXISTS ${SETTINGS_PATH})
     foreach (FILE_ENTRY ${FILE_ENTRIES})
         if ("${FILE_ENTRY}" MATCHES "^[^#]+=.*")
             string(REGEX MATCH "^[^=]+" SETTING_NAME ${FILE_ENTRY})
-            string(REGEX MATCH "[^=]+$" SETTING_VALUE ${FILE_ENTRY})
+            
+            # Allow for values to contain =
+            string(REGEX REPLACE "^[^=]+=(.*)" "\\1" SETTING_VALUE "${FILE_ENTRY}")
+            
             string(REPLACE "." ";" ENTRY_NAME_TOKENS ${SETTING_NAME})
             string(STRIP "${SETTING_VALUE}" SETTING_VALUE)
 
             list(LENGTH ENTRY_NAME_TOKENS ENTRY_NAME_TOKENS_LEN)
+            
+            # Some setting files contain settings such as "name"
+            # that come without separators
+            if("${ENTRY_NAME_TOKENS_LEN}" LESS 2)
```

Comment:
Some files can contain short names, e.g. the platform.txt file, e.g. 
https://github.com/arduino/Arduino/blob/cd9a15ebd1647336200b9328bc0912495e3a7195/hardware/arduino/avr/platform.txt#L8
This change ignores such lines.

Reply:
We're not reading from `platform.txt` only `boards.txt` and `programmers.txt`, just to clarify :)

Reply:
In know, but I once did and ran into problems with this. That's why I included it here, so others are not running into the same problems.

Reply:
@fraillt I believe that someday we'll support `platform.txt` as we've already discussed, so introducing this change early will only make out lives easier :)